### ISSUE TITLE
Should channel-less read*() routines be available by default?

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#19131]
Today, Chapel programs have access to write(), writef(), and writeln() for console output by default without having to `use`/`import` any modules.  As a Chapel programmer, I love this because I almost never write a program without using these routines—they feel like part and parcel of Chapel to me.  For the versions of these that support being called on a channel, `IO` must be used/imported, the argument being that they needed more of the IO machinery (channels), throw errors (so need the errors), etc.  In contrast the standalone versions halt on errors (essentially calling the channel versions with `try!`).

We did not do the same for the read(), readln(), readline(), readf() for reasons I don't quite recall.  Lydia suggests that it may have just been that we don't think of them as being used as often.  Another guess I had was that we think of them as being more erroneous by nature (more likely to hit bad input formats, e.g.) so more likely to throw, so more like the throwing channel.write*() routines.

Having just completed advent of code, however, which is very input heavy, I found myself questioning this decision and thinking that it would be more symmetrical (and convenient) to have the standalone `read*()` routines auto-available like the `write*()` versions.

I'm filing this issue to take the temperature on that idea.

**Comment by mppf** [comment]
Right, we made the ones available by default, like `writeln`, halt if there is an error, rather than throw. But in reading, it is much more likely one will find an error, e.g. if the file format did not match what you expected, or if there is a non-UTF-8 character, etc. So I think having `read()` be throwing seems important.

IMO it is not *that* common to use `stdin` and so it is reasonable to `use IO;` in programs that do.

**Comment by mstrout** [comment]
> For the versions of these that support being called on a channel, IO must be used/imported, the argument being that they needed more of the IO machinery (channels)

The top of the documentation on [ChapelIO](https://chapel-lang.org/docs/main/modules/standard/ChapelIO.html) says that that one dues not have to "use IO".  It also talks about some methods like readWriteThis() that might take a channel argument.  Is this a documentation bug, or am I misunderstanding something?

**Comment by bradcray** [comment]
@mstrout: There are two modules in play here:  `ChapelIO` (which we might want to call something like `AutoIO` if we end up going that convention for the part of `Math` that's auto-used and `IO`.  In this issue, I'm talking about moving some of the `read*()` routines into `ChapelIO`/`AutoIO` so that they're as symmetrically available as the `write()` cases.

One thing that I hadn't remembered when filing this is that some of the read() routines rely on throwing errors to signal things like EOF, which make it less like something you can just ignore (as with write() errors); that said I also found it somewhat confusing how different routines in the read* family used different things to indicate EOF, so have been meaning to review that to see if we're OK with it as well.  I'm currently planning on taking that on as a precondition to doing more here.

I would agree that console output is far more common to appear in programs than console input, but if it weren't for the throw-on-EOF issue, I'd still argue for including all standalone console I/O routines by default for symmetry.  The asymmetry seems worse to me than the argument that they're not commonly used.

**Comment by mppf** [comment]
Just linking to #18496 where we have been discussing if we need both the `read(type t)` and the `read(ref x)` forms. These have different EOF behavior (the first throws; the second returns false). It is not so clear to me what else we could do:
 * we will to throw on EOF for non-default-initializeable types (and currently we can access that with `read(type t)` when needed)
 * in that issue you indicated a pretty strong preference for keeping the bool-returning versions

Given all of that, I think we should stick with the status quo on read() routines not being available by default.
